src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SkippableFactAttribute.cs

```diff
@@ -30,10 +30,18 @@ public static void EvaluateSkips(bool skipOnNullOrWhiteSpaceEnv, bool skipOnTrue
                 setSkip($"Skipping because `{envName}` is null or whitespace");
                 break;
             }
-            else if (skipOnTrueEnv && bool.TryParse(envValue, out bool boolValue) && boolValue)
+            else if ((skipOnTrueEnv || skipOnFalseEnv) && bool.TryParse(envValue, out bool boolValue))
             {
-                setSkip($"Skipping because `{envName}` is set to True");
-                break;
+                if (skipOnTrueEnv && boolValue)
+                {
+                    setSkip($"Skipping because `{envName}` is set to True");
+                    break;
+                }
+                else if (skipOnFalseEnv && !boolValue)
```

Comment:
skipOnFalseEnv should also skip if the env is empty/not set; it really should skip anytime the value cannot be parsed as well as a false string.  The behavior should really be skipOnNotTrueEnv albeit that sounds strange.

Reply:
How about something like this? It will skip if `skipOnTrue` or `skipOnFalse` are set to invalid values.

```
else if ((skipOnTrueEnv || skipOnFalseEnv))
{
    if (bool.TryParse(envValue, out bool boolValue))
    {
        if ((skipOnTrueEnv && boolValue) || (skipOnFalseEnv && !boolValue))
        {
            setSkip($"Skipping because `{envName}` is set to {(boolValue ? "True" : "False")}");
            break;
        }
    }
    else
    {
        setSkip($"Skipping because `{envName}` is not a valid boolean");
        break;
    }
}
```

Reply:
I don't think that is the behavior we want for skipOnTrueEnv.  To me the way these are being used is important.  e.g. ExcludeOmniSharp vs IncludeArtifactsSize.  Those tests should only be excluded/included respectively with a true string.  Anything else should result in the default behavior.

Reply:
What is default behavior, specifically for skipOnTrueEnv? The way I'm thinking of the expected behavior is listed below:

**ExcludeOmniSharp & skipOnTrueEnv**
- ExcludeOmniSharp = True -> skip
- ExcludeOmniSharp = False -> run
- ExcludeOmniSharp = Invalid -> skip (should this be run in this case?)

**IncludeArtifactsSize & skipOnFalseEnv**
- IncludeArtifactsSize = True -> run
- IncludeArtifactsSize = False -> skip
- IncludeArtifactsSize = Invalid -> skip

Reply:
> What is default behavior, specifically for skipOnTrueEnv? 

If the ENV is not set or not set to true string the test will be run.  As I understand it, this change is altering that behavior.

Reply:
I see. [These changes](https://github.com/dotnet/installer/pull/17233/commits/1c7e46ebe08aa27eb14f51f6692e7c6d38d41734#diff-93670feecc27976c41bc9cc3803673cac717cf0c8e5e7f564f43f873d70ebce0R33) should implement that expected logic.